scargo/scargo_src/sc_debug.py

```diff
@@ -28,31 +28,37 @@ def __init__(self, config: Config, bin_path: Optional[Path]):
             )
             sys.exit(1)
 
-        self._bin_path = bin_path or self._get_bin_path(config.project.bin_name)
+        bin_name = config.project.bin_name
+        if not bin_name:
+            self._logger.error("No bin_name in config")
+            sys.exit(1)
+        self._bin_path = bin_path or self._get_bin_path(bin_name)
         if not self._bin_path.exists():
             self._logger.error("Binary %s does not exist", self._bin_path)
             self._logger.info("Did you run scargo build --profile Debug?")
             sys.exit(1)
 
         if self._target.family == "stm32":
-            self._chip = config.stm32.chip
+            stm32_config = config.stm32
```

Comment:
I don't think pydantic would allow stm32_config to be None or something so I don't see reason to do this.

Reply:
The [stm32] section is optional (and not present in non-stm projects, so it *has* to be optional)
```stm32: Optional["Stm32Config"]```
When the section is not present, it will be `None`.

Reply:
Okay, then this needs to be reworked in sc_config.py. If config.project.target is stm32 than Stm32 section is mandatory.

Reply:
As mentioned in another comment, this is completely separate from typing, and so it's out of scope of this PR.
Keep in mind that as long as we have one config class (and I don't really see a clean way to have more classes), some kind of reassurance for mypy will still be needed.

Reply:
Another way might be a method like `Config.get_stm32_config()` that will throw the error if it's not there.

Reply:
I've added validation